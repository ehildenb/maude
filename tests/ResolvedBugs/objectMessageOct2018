#!/bin/sh

MAUDE_LIB=$srcdir/../../src/Main
export MAUDE_LIB

../../src/Main/maude \
  < $srcdir/objectMessageOct2018.maude -no-banner -no-advise \
  > objectMessageOct2018.out 2>&1

diff $srcdir/objectMessageOct2018.expected objectMessageOct2018.out > /dev/null 2>&1
