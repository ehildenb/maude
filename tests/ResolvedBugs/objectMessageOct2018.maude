set show timing off .
set show advisories off .

***
***	Check that object-message rewriting fake rewrites are ignored
***	by trace/break/profile code.
***

mod FOO is
  inc CONFIGURATION .
  op User : -> Cid .
  ops me other : -> Oid .
  op no-op : Oid Oid -> Msg [msg] .

vars X Y : Oid .
var AS : AttributeSet .
  rl < X : User | AS >  no-op(X, Y) => < X : User | AS > .
endm

set break on .
break select __ .

set trace on .
set trace whole on .
set profile on .

frew < me : User | none > no-op(me, other) .

*** we expect a single break to debugger
resume .

show profile .
