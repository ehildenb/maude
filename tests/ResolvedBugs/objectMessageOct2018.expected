==========================================
frewrite in FOO : no-op(me, other) < me : User | none > .
break on symbol: __
*********** rule
rl no-op(X, Y) < X : User | AS > => < X : User | AS > .
X --> me
Y --> other
AS --> (none).AttributeSet
Old: no-op(me, other) < me : User | none >
no-op(me, other) < me : User | none >
--->
< me : User | none >
New: < me : User | none >
rewrites: 1
result Object: < me : User | none >
rl no-op(X, Y) < X : User | AS > => < X : User | AS > .
rewrites: 1 (100%)

Bye.
